sooyoung/services/userService.js

```diff
@@ -10,9 +12,36 @@ const signUp = async (username, email, profileImage, password) => {
     err.statusCode = 400;
     throw err;
   }
-  return await userDao.createUser(username, email, profileImage, password);
+
+  const COST_FACTOR = 12;
+  const hashedPassword = await bcrypt.hash(password, COST_FACTOR);
+
+  return await userDao.createUser(
+    username,
+    email,
+    profileImage,
+    hashedPassword
+  );
+};
+
+const signIn = async (email, password) => {
+  const hashedPassword = await userDao.getHashedPassword(email);
+  const isMatchPassword = await bcrypt.compare(password, hashedPassword);
+
+  if (!isMatchPassword) {
+    const err = new Error('PASSWORD_IS_NOT_MATCHED');
+    err.statusCode = 401;
+    throw err;
+  }
+
+  const getUserId = await userDao.getUserId(email);
+  const secretKey = process.env.SECRET_KEY;
+  const jwtToken = jwt.sign({ userId: getUserId }, secretKey);
+
+  return jwtToken;
```

Comment:
```suggestion

  const userId = await userDao.getUserId(email);

  return jwt.sign({ userId }, process.env.SECRET_KEY);
```
1. userDao.getUserId(email)을 해서 return되는 값(userId)이 변수에 담기는 것이기 때문에 getUserId보다는 값 자체를 의미하는 userId가 더 자연스러울 것 같습니다.
2. 복잡하고 가독성에 지장이 있는 코드가 아니기떄문에 다음과 같이 곧바로 리턴하면 4줄로 작성된 코드를 절반을 줄일 수 있습니다.